opentelemetry-api/src/trace/tracer.rs

```diff
@@ -325,7 +326,7 @@ impl SpanBuilder {
     }
 
     /// Assign span attributes
-    pub fn with_attributes(self, attributes: Vec<KeyValue>) -> Self {
```

Comment:
Thanks for the change. It looks better. But do you think we can do something like 
```rust
    pub fn with_attributes<I: IntoIterator<Item = KeyValue>>(self, attributes: I) -> Self {
        SpanBuilder {
            attributes: Some(OrderMap::from_iter(attributes)),
            ..self
        }
    }
```
This way we won't break the existing implementation and also hide the `OrderMap` from the end user, who probably don't need to know this implementation details. 

(I know this PR is a breaking change anyway but I think most of the end user will only set the attributes here so maybe we can make their upgrade easier)

maybe add another function `with_attributes_order_map` which accepts `OrderMap<Key,Value>` as parameters.

Reply:
That's a great shout - I've amended it!